DQM: Remove more DQMStore features

schneiml opened the pull request:
#### PR description:
This PR removes more rarely used functionality from the DQM infrastructure, namely
- `scaleElements`: not sure what this was ever used for, online maybe?
- `softReset`: this feature is rather useful, but the few usages don't really justify the complexity. Used mainly in ECAL DQM.
- `setAccumulate` flag: basically unused, not sure what it was supposed to signal.
- `remove*`: Used in a bunch of places mostly for non-standard workflows. Remove is very hard to support safely (even today, it might randomly invalidate pointers that other modules are holding), so we have to drop it. The most common pattern is *"check if ME exists, if yes remove, then book"*: this probably was not required in most cases, but it might be required to re-bin plots. We might change semantics of `book*` to *replace* an existing ME if *merge* is not possible, without invalidating the exisiting MEs (only replace the `TH1`). [update]
- `QStatisticalTests`: completely unused. [update]

#### PR validation:
Requires feedback from subsystem users, mainly ECAL (@tanmaymudholkar ?) regarding `softReset`. I suspect this PR will change behavior for ECAL online DQM.

The other user of `softReset` is `HLXMonitor`, which as far as I understand is history. Not sure if anybody feels responsible.

`scaleElements` was called in the `MEtoEDM` workflow, so it might affect AlCa/PCL. I suspect it does not (I can't see any configuration for it, and I don't see what it would be good for), but it might be worth another check.

Remove was used in a lot of places, I suspect in most cases this will only make additional MEs show up in the output. However, there might be cases where re-booking now behaves incorrectly. I typically removed the entire `cleanup` functionality incl. config options, unless that seemed to complicated, then I left a comment in the now-useless function stubs.

schneiml committed:
Remove scaleElements.

Not sure what it was supposed to do, the single usage was probably a
no-op as I can't see any config for the scale factor.

schneiml committed:
Remove softReset and related.

The main user is ECAL; I suspect the feature is actually used in ECAL
online DQM. The other big user is HLX.

I didn't bother to remove all of the implementation, for now getting rid
of the APIs is the main point.

schneiml committed:
Remove remaining uses of Accumulate flag.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28247/12393

- This PR adds an extra 108KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/EcalMonitorTasks/plugins/EcalDQMonitorTask.cc  modified in PR(s): #28099
  - File  DQM/EcalPreshowerMonitorModule/src/ESIntegrityTask.cc  modified in PR(s): #28099
  - File  DQM/TrackingMonitor/interface/TrackAnalyzer.h  modified in PR(s): #27677
  - File  DQM/TrackingMonitor/src/TrackAnalyzer.cc  modified in PR(s): #27677
  - File  DQMServices/Core/interface/DQMStore.h  modified in PR(s): #28092
  - File  DQMServices/Core/interface/MonitorElement.h  modified in PR(s): #28092
  - File  DQMServices/Core/src/DQMStore.cc  modified in PR(s): #28092
  - File  DQMServices/Core/src/MonitorElement.cc  modified in PR(s): #28092

cmsbuild commented:
A new Pull Request was created by  @schneiml (Marcel Schneider) for master.

It involves the following packages:

DQM/EcalCommon
DQM/EcalMonitorTasks
DQM/EcalPreshowerMonitorModule
DQM/HLXMonitor
DQM/TrackingMonitor
DQMServices/Components
DQMServices/Core


@andrius-k, @kmaeshima, @schneiml, @cmsbuild, @jfernan2, @fioriNTU can you please review it and eventually sign? Thanks.
@barvic, @hdelanno, @makortel, @mtosi, @argiro, @missirol, @fioriNTU, @jandrea, @idebruyn, @threus this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

schneiml commented:
please test

I don't expect the PR tests to be very useful for validation here, but let's see if they notice anything.

cmsbuild commented:
+1
Tested at: faf2c5253a75a2b6287a8ea8336cb1f0354fc423
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3114/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3114/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 10 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2961036
 - DQMHistoTests: Total failures: 128
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2960567
 - DQMHistoTests: Total skipped: 341
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

schneiml commented:
No significant changes in the comparison. Still requires confirmation from ECAL and AlCa.

schneiml committed:
Remove DumpFile, MergeFile and DQMCollector.

While probably useful, I don't think these tools are used much today.
We cannot support these heavily 'DQMstore as a Service' based tools in the future, so let's remove them already and see.
DQMCollector is probably actively used in the GUI, but this is only the launcher, not the actual DQMNet implementation.
Also the GUI uses a fixed version of these files from CMSSW7_6_0, it is effectively diverged.

schneiml committed:
Remove ReadFileExample.

Reading files into DQMStore is no longer supported, because global DQMStore itself is on the way out.

schneiml committed:
Remove deleteAfterCopy from MEtoEDM.

... because we will remove DQMStore deletion entirely.

schneiml committed:
Remove removeModuleLevelHists.

schneiml committed:
Remove removeElement in BeamMonitor.

If this moulde actually needs re-binning, we need to find another solution.
Probably we'll just have re-booking replace the existing histo.

schneiml committed:
Remove cleanup option from ECAL.

schneiml committed:
Remove 'booked already' checks from L1T

schneiml committed:
Remove more random remove calls from subsystem code.

schneiml committed:
Remove the remove APIs.

schneiml committed:
Code-format.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28247/12412

- This PR adds an extra 324KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/EcalMonitorTasks/plugins/EcalDQMonitorTask.cc  modified in PR(s): #28099
  - File  DQM/EcalPreshowerMonitorModule/src/ESIntegrityTask.cc  modified in PR(s): #28099
  - File  DQM/Physics/test/topDQM_production_cfg.py  modified in PR(s): #28156
  - File  DQM/TrackingMonitor/interface/TrackAnalyzer.h  modified in PR(s): #27677
  - File  DQM/TrackingMonitor/src/TrackAnalyzer.cc  modified in PR(s): #27677
  - File  DQMServices/Core/interface/DQMStore.h  modified in PR(s): #28092
  - File  DQMServices/Core/interface/MonitorElement.h  modified in PR(s): #28092
  - File  DQMServices/Core/src/DQMStore.cc  modified in PR(s): #28092
  - File  DQMServices/Core/src/MonitorElement.cc  modified in PR(s): #28092
  - File  DQMServices/Core/test/BuildFile.xml  modified in PR(s): #28092

cmsbuild commented:
Pull request #28247 was updated. @cmsbuild, @andrius-k, @kmaeshima, @christopheralanwest, @fwyzard, @schneiml, @tocheng, @Martin-Grunewald, @franzoni, @jfernan2, @fioriNTU, @tlampen, @pohsun can you please check and sign again.

schneiml commented:
please test

Now we expect changes, mainly a few additional MEs. However, we need to pay special attention to any *changes* caused by the last update.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 362c74e4f8ee74e9ef2363e24508e0545ad8c7db
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3126/summary.html

cmsbuild commented:
Comparison job queued.

schneiml committed:
Remove unused Statistical helpers.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3126/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2961036
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2960694
 - DQMHistoTests: Total skipped: 341
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1530.021 KiB( 33 files compared)
 - DQMHistoSizes: changed ( 1000.0,... ): 25.259 KiB       Egamma/gedPhotonAnalyzer
 - DQMHistoSizes: changed ( 1000.0,... ): 25.259 KiB       Egamma/stdPhotonAnalyzer
 - DQMHistoSizes: changed ( 1000.0,... ): 5.613 KiB        Egamma/Electrons
 - DQMHistoSizes: changed ( 10024.0,... ): 2.364 KiB        BDHadronTracks/JetContent
 - DQMHistoSizes: changed ( 10024.0,... ): 0.842 KiB        ParticleFlow/PFJetValidation
 - DQMHistoSizes: changed ( 10024.0,... ): 0.451 KiB        ParticleFlow/PFMETValidation
 - DQMHistoSizes: changed ( 20034.0,... ): 25.649 KiB       Egamma/stdPhotonAnalyzerHGCalFromMultiCl
 - DQMHistoSizes: changed ( 20034.0,... ): 7.058 KiB        Egamma/Electrons
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

schneiml commented:
It looks like the only changes visible in the tests is a moderate number of new plots, most in the `Egamma` hierarchy. They are probably not useful (else, why would they have been deleted?), but should also not hurt.

I'll add another few non-risky changes to this PR (removing some unused files) and then rebase it now that #28099 is in.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28247/12424

- This PR adds an extra 460KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/Physics/test/topDQM_production_cfg.py  modified in PR(s): #28156
  - File  DQM/TrackingMonitor/interface/TrackAnalyzer.h  modified in PR(s): #27677
  - File  DQM/TrackingMonitor/src/TrackAnalyzer.cc  modified in PR(s): #27677
  - File  DQMServices/Core/interface/DQMStore.h  modified in PR(s): #28092
  - File  DQMServices/Core/interface/MonitorElement.h  modified in PR(s): #28092
  - File  DQMServices/Core/src/DQMStore.cc  modified in PR(s): #28092
  - File  DQMServices/Core/src/MonitorElement.cc  modified in PR(s): #28092
  - File  DQMServices/Core/test/BuildFile.xml  modified in PR(s): #28092


Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-checks**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28247/12424/code-checks.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28247/12424/code-checks.patch | patch -p1`
You can also run `scram build code-checks` to apply code checks directly

cmsbuild commented:
The code-checks are being triggered in jenkins.

schneiml commented:
rebased to yesterday's IB to resolve the merge conflicts (which in turn failed the code checks)

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28247/12468

- This PR adds an extra 376KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/EcalPreshowerMonitorModule/src/ESIntegrityTask.cc  modified in PR(s): #28092
  - File  DQM/Physics/test/topDQM_production_cfg.py  modified in PR(s): #28156
  - File  DQM/TrackingMonitor/interface/TrackAnalyzer.h  modified in PR(s): #27677
  - File  DQM/TrackingMonitor/src/TrackAnalyzer.cc  modified in PR(s): #27677
  - File  DQMServices/Core/interface/DQMStore.h  modified in PR(s): #28092
  - File  DQMServices/Core/interface/MonitorElement.h  modified in PR(s): #28092
  - File  DQMServices/Core/src/DQMStore.cc  modified in PR(s): #28092
  - File  DQMServices/Core/src/MonitorElement.cc  modified in PR(s): #28092
  - File  DQMServices/Core/test/BuildFile.xml  modified in PR(s): #28092

cmsbuild commented:
Pull request #28247 was updated. @cmsbuild, @andrius-k, @kmaeshima, @christopheralanwest, @fwyzard, @schneiml, @tocheng, @Martin-Grunewald, @franzoni, @jfernan2, @fioriNTU, @tlampen, @pohsun can you please check and sign again.

schneiml commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 772835d5e62fd94821dd92637d151a26d3c36df5
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3190/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #28205 @bsunanda: Run3-hcx232 Attempt to remove access of DDCompactView from SimG4CMS/Calo code (Step 2)
 - #28270 @mrodozov: Fix gcc9 warning in DetectorDescription/DDCMS (missing struct initial
 - #28236 @bsunanda: Phase2-hgx226 Correct wafer size for the HGCal V11 version
 - #28212 @cericeci: Propagate inverse beta variable to pat muons (muon POG)
 - #28257 @kpedro88: Include dedx collections in fastsim output

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3190/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3190/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3190/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3107 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2961036
 - DQMHistoTests: Total failures: 5082
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2955613
 - DQMHistoTests: Total skipped: 341
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1530.021 KiB( 33 files compared)
 - DQMHistoSizes: changed ( 1000.0,... ): 25.259 KiB       Egamma/gedPhotonAnalyzer
 - DQMHistoSizes: changed ( 1000.0,... ): 25.259 KiB       Egamma/stdPhotonAnalyzer
 - DQMHistoSizes: changed ( 1000.0,... ): 5.613 KiB        Egamma/Electrons
 - DQMHistoSizes: changed ( 10024.0,... ): 2.364 KiB        BDHadronTracks/JetContent
 - DQMHistoSizes: changed ( 10024.0,... ): 0.842 KiB        ParticleFlow/PFJetValidation
 - DQMHistoSizes: changed ( 10024.0,... ): 0.451 KiB        ParticleFlow/PFMETValidation
 - DQMHistoSizes: changed ( 20034.0,... ): 25.649 KiB       Egamma/stdPhotonAnalyzerHGCalFromMultiCl
 - DQMHistoSizes: changed ( 20034.0,... ): 7.058 KiB        Egamma/Electrons
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

Martin-Grunewald commented:
+1

fabiocos commented:
please test

with a clean reference

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 772835d5e62fd94821dd92637d151a26d3c36df5
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3264/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3264/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2961036
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2960694
 - DQMHistoTests: Total skipped: 341
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1530.021 KiB( 33 files compared)
 - DQMHistoSizes: changed ( 1000.0,... ): 25.259 KiB       Egamma/gedPhotonAnalyzer
 - DQMHistoSizes: changed ( 1000.0,... ): 25.259 KiB       Egamma/stdPhotonAnalyzer
 - DQMHistoSizes: changed ( 1000.0,... ): 5.613 KiB        Egamma/Electrons
 - DQMHistoSizes: changed ( 10024.0,... ): 2.364 KiB        BDHadronTracks/JetContent
 - DQMHistoSizes: changed ( 10024.0,... ): 0.842 KiB        ParticleFlow/PFJetValidation
 - DQMHistoSizes: changed ( 10024.0,... ): 0.451 KiB        ParticleFlow/PFMETValidation
 - DQMHistoSizes: changed ( 20034.0,... ): 25.649 KiB       Egamma/stdPhotonAnalyzerHGCalFromMultiCl
 - DQMHistoSizes: changed ( 20034.0,... ): 7.058 KiB        Egamma/Electrons
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

schneiml commented:
+1

We still have not heard from ECAL, but they where informed and we decided to move on anyways.

christopheralanwest commented:
@adewit @mmusich Could one of you provide feedback on the tracker and tracker alignment changes in the packages in the AlCa domain?

schneiml commented:
I will rebase this one once it is fully signed or #28092 gets merged. Not much point in the excercise otherwise. (Or once I need to for some other reason).

fabiocos commented:
@christopheralanwest @adewit @mmusich as we need to finally converge here (after closing #28092, which I plan to do in next IB), could you please express yourselves asap on this proposal?

adewit commented:
Sorry, I missed this before. I don't think the changes in Alignment/OfflineValidation would break any workflows, so fine for me.

christopheralanwest commented:
+1

(aside from the need to rebase)

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

mmusich commented on Calibration/TkAlCaRecoProducers/python/ALCARECOPromptCalibProdSiStripGains_cff.py:
```diff
@@ -99,7 +99,6 @@
                                             # 2 provide more detailed output
                                             Frequency = cms.untracked.int32(50),
                                             MEPathToSave = cms.untracked.string('AlCaReco/SiStripGains'),
-                                            deleteAfterCopy = cms.untracked.bool(True)
```
on a different topic, is this supposed to be useless with the new DQM infrastructure?
Looking back in my inbox this seemed to be necessary not to explode with memory at Tier-0 during operations (but this was back in 2016).

mmusich replied:
cf: https://github.com/cms-sw/cmssw/pull/14198#issuecomment-214433875

schneiml replied:
I am not exactly sure what this workflow does, but in general the memory concerns at Tier0 are pretty much gone since we no longer use per-stream MEs.

If the copy in this module (which hopefully does run in a  HARVESTING step, and is not affected at all by the stream story) consumes too much memory, we need to find a different solution (like, book things in the right place, or read them from the right place, or rename things after the fact), but moving around stuff in the DQMStore is not something we can support for the future.

Actually, the comment previous to the one you linked mentioned that this is a end-of-job spike, which from legends I know was one of the reasons to move towards the DQMIO format. I wonder if we should finally stop pretending to this code that it is still run 1 and move it to current infrastructure... Sadly, I have no real clue how these workflows work and it seems nobody else has either, so that will be hard.

mmusich replied:
@schneiml 
> the memory concerns at Tier0 are pretty much gone since we no longer use per-stream MEs.

I am not sure how relevant it is, as to the best of my knowledge Tier0 AlCaSkim and AlCaHarvest jobs are single-threaded.
I would feel much better if we could organize (for this PR and the others that will come that will change `MEtoEDMConverter`) a more meaningful check than 10 events in the PR test.  For some workflows a 10 event test will probe nothing as the computation won't even be started.
Can AlCa/DB conveners make sure this happens? @christopheralanwest @tlampen @tocheng 

>  I wonder if we should finally stop pretending to this code that it is still run 1 and move it to current infrastructure...

I would be happy to take a look to achieve this. Can you explain what you mean by "current" infrastructure?

schneiml replied:
"current": Not `MEtoEDM`, if possible, or otherwise a good reasoning why it is required, and no reliance on `edm::Service<DQMStore>` (which implies using `DQMEDAnalyzer` or `DQMEDHarvester` base classes). It is well possible that this is not possible for some reason at the moment, but then we need to find out these reasons and decide what to do for the future.

mmusich replied:
so, I took the liberty of doing a bit of profiling. 
I've re-run a similar step of what I had done here https://github.com/cms-sw/cmssw/pull/22515 i.e.:
```
cmsDriver.py step3 --datatier ALCARECO --conditions auto:run2_data -s ALCA:PromptCalibProdSiStripGains --eventcontent ALCARECO -n 1000 --dasquery='file dataset=/ZeroBias/Run2016C-SiStripCalMinBias-18Apr2017-v1/ALCARECO run=276097'
```
and profiled the RSS memory:

![memory](https://user-images.githubusercontent.com/5082376/68586869-f113f580-0485-11ea-9892-058419302340.png)

The spike at the end of the job is increased indeed, but I cannot comment if that is going to be a problem for Tier-0.
I also checked the output of the harvesting job, tweaking the configuration obtained via:

```
cmsDriver.py stepHarvest --data --conditions auto:run2_data --scenario pp -s ALCAHARVEST:SiStripGains --filein file:PromptCalibProdSiStripGains.root -n 1000 --fileout file:calib.root
```
in order to be forced to produce a payload via:
```python
process.alcaSiStripGainsHarvester.GoodFracForTagProd =  cms.untracked.double(0.0)   #<- fraction of good fits (default is 98%)
process.alcaSiStripGainsHarvester.NClustersForTagProd = cms.untracked.double(100000)   #<- total number of clusters (default used is 8e8)
```

and the good news is that the payload hash is the same and I don't see differences in the harvested output (full diff  [here](http://musich.web.cern.ch/musich/display/compare_SiStripGains_28247.pdf) ):
![image](https://user-images.githubusercontent.com/5082376/68587628-cb87eb80-0487-11ea-9cfc-39fb25d4d44b.png)

schneiml replied:
Thanks @mmusich for the test!

The observation is, I'd say, exactly compatible with what is done here: During processing, the memory use is the same, but the spike while `MEtoEDM` saving is higher if the unnecessary MEs are not deleted. The absolute change seems not too big to me, not sure if Tier0 sees this the same.

That the produced payload is identical is good to know.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28247/12703

- This PR adds an extra 320KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/EcalPreshowerMonitorModule/src/ESIntegrityTask.cc  modified in PR(s): #28092, #28342
  - File  DQM/TrackingMonitor/interface/TrackAnalyzer.h  modified in PR(s): #27677
  - File  DQM/TrackingMonitor/src/TrackAnalyzer.cc  modified in PR(s): #27677, #28342
  - File  DQMOffline/EGamma/src/ElectronDqmHarvesterBase.cc  modified in PR(s): #28342
  - File  DQMOffline/PFTau/plugins/PFClient.cc  modified in PR(s): #28342
  - File  DQMServices/ClientConfig/interface/DQMGenericClient.h  modified in PR(s): #28316
  - File  DQMServices/Core/interface/DQMStore.h  modified in PR(s): #28092, #28316, #28342
  - File  DQMServices/Core/interface/MonitorElement.h  modified in PR(s): #28092, #28316, #28342
  - File  DQMServices/Core/src/DQMStore.cc  modified in PR(s): #28092, #28316, #28342
  - File  DQMServices/Core/src/MonitorElement.cc  modified in PR(s): #28092, #28316, #28342
  - File  DQMServices/Core/test/BuildFile.xml  modified in PR(s): #28092, #28342

cmsbuild commented:
Pull request #28247 was updated. @cmsbuild, @andrius-k, @kmaeshima, @christopheralanwest, @fwyzard, @schneiml, @tocheng, @Martin-Grunewald, @franzoni, @jfernan2, @fioriNTU, @tlampen, @pohsun can you please check and sign again.

schneiml commented:
please test

Rebased to CMSSW_11_0_X_2019-11-10-2300. There where more problems than expected (random build failures), but a clean build seems to be ok. Let's see what the bot thinks.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: fe8e8523b8e2d497375680a9e312a40383e50abf
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3427/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3427/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2936360
 - DQMHistoTests: Total failures: 21
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2935998
 - DQMHistoTests: Total skipped: 341
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1454.244 KiB( 33 files compared)
 - DQMHistoSizes: changed ( 1000.0,... ): 25.259 KiB       Egamma/gedPhotonAnalyzer
 - DQMHistoSizes: changed ( 1000.0,... ): 25.259 KiB       Egamma/stdPhotonAnalyzer
 - DQMHistoSizes: changed ( 1000.0,... ): 5.613 KiB        Egamma/Electrons
 - DQMHistoSizes: changed ( 10024.0,... ): 2.364 KiB        BDHadronTracks/JetContent
 - DQMHistoSizes: changed ( 10024.0,... ): 0.842 KiB        ParticleFlow/PFJetValidation
 - DQMHistoSizes: changed ( 10024.0,... ): 0.451 KiB        ParticleFlow/PFMETValidation
 - DQMHistoSizes: changed ( 20034.0,... ): 25.649 KiB       Egamma/stdPhotonAnalyzerHGCalFromMultiCl
 - DQMHistoSizes: changed ( 20034.0,... ): 7.058 KiB        Egamma/Electrons
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

schneiml commented:
+1

Everything looks as before.

fabiocos commented:
@christopheralanwest @tlampen @Martin-Grunewald could you please verify the rebase and sign it in case?

christopheralanwest commented:
@fabiocos Was the question about whether the observed increase in RSS memory in https://github.com/cms-sw/cmssw/pull/28247/files#r344690786 is acceptable to Tier-0 ever answered? Given that the RSS memory in the test job was previously only 1.0 GB, the 15% increase in RSS memory usage doesn't sound problematic, but I don't know whether there are other jobs in which a 15% memory increase would be undesirable.

Martin-Grunewald commented:
+1

fabiocos commented:
@drkovalskyi @davidlange6 concerning the question about possible impact in T0 of the increase in RSS for ALCARECO jobs implied by this PR, as discussed in https://github.com/cms-sw/cmssw/pull/28247#pullrequestreview-314196836 , could you please comment whether the observed effect can be easily absorbed by standard operational conditions?

davidlange6 commented:
hi @fabiocos - the question is pretty out of context... 150MB increase seen only in jobs that take 1GB RSS is indeed presumably nothing to be worried about.

tlampen commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

mmusich commented:
@fabiocos, all, about:

> possible impact in T0 of the increase in RSS for ALCARECO jobs implied by this PR, as discussed in #28247 (review).

I just want to point out I tested only one of the PCL workflows, so the overall increase might be more significant than 150MB.

fabiocos commented:
@davidlange6 that might be my naive conclusion, but I wanted to hear from past historical memory when the problem occurred, also based on the memory allocation for these jobs at T0.

drkovalskyi commented:
I don't remember us being at the limit of memory usage for Express jobs at Tier0. Given that total memory usage depends on the configuration of Tier0, i.e. what ALCA/PCL we run we won't be able to give an exact answer without a replay with a realistic configuration.

fabiocos commented:
@drkovalskyi I think such a test would be welcome once this code is integrated inside the release, from the standalone tests seen in this review we have no evidence of obvious disasters I would say.

drkovalskyi commented:
Yes, we can do it. We created a ticket to follow up on it at https://its.cern.ch/jira/browse/CMSTZ-836

davidlange6 commented:
Because we no longer know the interesting list of alcas for the tier0 configuration and thus can’t do a test locally? (Will be hard to revert this in a month I guess)

On Nov 19, 2019, at 4:07 PM, Fabio Cossutti <notifications@github.com<mailto:notifications@github.com>> wrote:


@drkovalskyi<https://github.com/drkovalskyi> I think such a test would be welcome once this code is integrated inside the release, from the standalone tests seen in this review we have no evidence of obvious disasters I would say.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/28247?email_source=notifications&email_token=ABGPFQ5XWOPQWHELWYQQ4N3QUP6M7A5CNFSM4JEB265KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEEOQJTY#issuecomment-555549903>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGPFQY4GIEJ7YFUSPRJHYLQUP6M7ANCNFSM4JEB265A>.

drkovalskyi commented:
I don't expect problems if we have 10-15% overall increase in memory usage for Express. A replay is just a safety check. If someone wants to do a test now, here is Tier0 Express config for one of the last big collisions runs: https://cmsweb.cern.ch/t0wmadatasvc/prod/express_config?run=325170.

fabiocos commented:
code-checks

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28247/12838

- This PR adds an extra 424KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/OfflineValidation/plugins/TrackerOfflineValidationSummary.cc  modified in PR(s): #28380, #28379, #28407
  - File  Alignment/OfflineValidation/python/TkAlAllInOneTool/offlineValidationTemplates.py  modified in PR(s): #28380, #28379, #28407
  - File  Alignment/OfflineValidation/python/TrackerOfflineValidationSummary_cfi.py  modified in PR(s): #28380, #28379, #28407
  - File  Alignment/OfflineValidation/python/TrackerOfflineValidation_Dqm_cff.py  modified in PR(s): #28380, #28379, #28407
  - File  Alignment/OfflineValidation/test/dqmOfflinevalidator_cfg.py  modified in PR(s): #28380, #28379, #28407
  - File  Calibration/EcalCalibAlgos/python/ALCARECOPromptCalibProdEcalPedestals_cff.py  modified in PR(s): #28380, #28379, #28407
  - File  Calibration/TkAlCaRecoProducers/python/ALCARECOPromptCalibProdSiStripGainsAAG_cff.py  modified in PR(s): #28380, #28379, #28407
  - File  Calibration/TkAlCaRecoProducers/python/ALCARECOPromptCalibProdSiStripGains_cff.py  modified in PR(s): #28380, #28379, #28407
  - File  Calibration/TkAlCaRecoProducers/python/ALCARECOSiStripPCLHistos_cff.py  modified in PR(s): #28380, #28379, #28407
  - File  Calibration/TkAlCaRecoProducers/test/convertSiStripRootDQMFileToEDM_cfg.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/BeamMonitor/plugins/AlcaBeamMonitorClient.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/BeamMonitor/plugins/BeamMonitorBx.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalCommon/interface/MESet.h  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalCommon/interface/MESetDet2D.h  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalCommon/src/MESet.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalCommon/src/MESetDet2D.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalMonitorDbModule/src/MonitorElementsDb.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalMonitorTasks/interface/DQWorkerTask.h  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalMonitorTasks/plugins/EcalDQMonitorTask.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalMonitorTasks/src/DQWorkerTask.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalPreshowerMonitorClient/python/EcalPreshowerLocalMonitorClient_cfi.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalPreshowerMonitorModule/interface/ESDaqInfoTask.h  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalPreshowerMonitorModule/interface/ESDataCertificationTask.h  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalPreshowerMonitorModule/interface/ESDcsInfoTask.h  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalPreshowerMonitorModule/python/ESDaqInfoTask_cfi.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalPreshowerMonitorModule/python/ESDataCertificationTask_cfi.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalPreshowerMonitorModule/python/ESDcsInfoTask_cfi.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalPreshowerMonitorModule/src/ESDaqInfoTask.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalPreshowerMonitorModule/src/ESDataCertificationTask.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalPreshowerMonitorModule/src/ESDcsInfoTask.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/EcalPreshowerMonitorModule/src/ESIntegrityTask.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/HLTEvF/plugins/FourVectorHLT.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/HLXMonitor/interface/HLXMonitor.h  modified in PR(s): #28380, #28379, #28407
  - File  DQM/HLXMonitor/python/hlx_dqm_sourceclient_cfi.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/HLXMonitor/python/hlx_dqm_sourceclient_test_cfi.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/HLXMonitor/python/hlx_dqm_sourceclient_vme22_cfi.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/HLXMonitor/src/HLXMonitor.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/HLXMonitor/test/hlx_dqm_sourceclient_cfi.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/L1TMonitorClient/src/L1EmulatorErrorFlagClient.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/L1TMonitorClient/src/L1TEMTFEventInfoClient.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/L1TMonitorClient/src/L1TEventInfoClient.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/Physics/test/PFJet_topDQM_production_cfg.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/Physics/test/topDQM_production_PU_cfg.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/Physics/test/topDQM_production_cfg.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/Physics/test/topDQM_production_miniAOD.py  modified in PR(s): #28380, #28379, #28407
  - File  DQM/SiStripCommissioningClients/src/CommissioningHistograms.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/SiStripCommissioningSources/src/SiStripCommissioningSource.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQM/TrackingMonitor/interface/TrackAnalyzer.h  modified in PR(s): #28380, #28379, #28407, #27677
  - File  DQM/TrackingMonitor/src/TrackAnalyzer.cc  modified in PR(s): #28380, #28379, #28407, #27677
  - File  DQMOffline/EGamma/plugins/PhotonOfflineClient.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQMOffline/EGamma/src/ElectronDqmHarvesterBase.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQMOffline/PFTau/plugins/PFClient.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQMOffline/PFTau/plugins/PFClient_JetRes.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQMOffline/Trigger/plugins/DQMOfflineHLTEventInfoClient.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/ClientConfig/interface/DQMGenericClient.h  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Components/plugins/DQMMessageLoggerClient.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Components/plugins/MEtoEDMConverter.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Components/plugins/MEtoEDMConverter.h  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Components/python/MEtoEDMConverter_cfi.py  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/bin/BuildFile.xml  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/bin/DQMCollector.cpp  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/bin/DumpFile.cpp  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/bin/MergeFile.cpp  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/interface/DQMStore.h  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/interface/MonitorElement.h  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/src/DQMStore.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/src/MonitorElement.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/src/QStatisticalTests.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/src/QStatisticalTests.h  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/src/QTest.cc  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/test/BuildFile.xml  modified in PR(s): #28380, #28379, #28407
  - File  DQMServices/Core/test/DQMReadFileExample.cc  modified in PR(s): #28380, #28379, #28407
  - File  HLTrigger/Timer/plugins/ThroughputServiceClient.cc  modified in PR(s): #28380, #28379, #28407
  - File  HLTriggerOffline/B2G/test/b2gHLTDQM_production_cfg.py  modified in PR(s): #28380, #28379, #28407
  - File  Validation/RecoB/plugins/BDHadronTrackMonitoringHarvester.cc  modified in PR(s): #28380, #28379, #28407
  - File  Validation/RecoTau/plugins/dqmAuxFunctions.cc  modified in PR(s): #28380, #28379, #28407

fabiocos commented:
please test workflow 1010.0,1020.0,1030.0,1040.0

cmsbuild commented:
The tests are being triggered in jenkins.

fabiocos commented:
@schneiml in my private tests based on CMSSW_11_0_X_2019-11-19-2300 I see that 1040.0 is now crashing...

schneiml commented:
@fabiocos 1040.0 runs for me, after clean rebase on `CMSSW_11_0_X_2019-11-19-2300`. Let's see what the bot thinks...

I have not checked the output, though.

cmsbuild commented:
+1
Tested at: fe8e8523b8e2d497375680a9e312a40383e50abf
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3551/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-734c5c/3551/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-734c5c/1010.0_TestEnableEcalHCAL2017B+TestEnableEcalHCAL2017B+TIER0EXPTE+ALCAEXPTE+ALCAHARVDTE
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-734c5c/1020.0_AlCaLumiPixels2016H+AlCaLumiPixels2016H+TIER0EXPLP+ALCAEXPLP+ALCAHARVLP+TIER0PROMPTLP
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-734c5c/1030.0_RunHLTPhy2017B+RunHLTPhy2017B+TIER0EXPHPBS+ALCASPLITHPBS+ALCAHARVDHPBS+ALCAHARVDHPBSLOWPU
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-734c5c/1040.0_RunZeroBias2017F+RunZeroBias2017F+TIER0RAWSIPIXELCAL+ALCASPLITSIPIXELCAL+ALCAHARVDSIPIXELCAL

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2936360
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2936017
 - DQMHistoTests: Total skipped: 341
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1454.244 KiB( 33 files compared)
 - DQMHistoSizes: changed ( 1000.0,... ): 25.259 KiB       Egamma/gedPhotonAnalyzer
 - DQMHistoSizes: changed ( 1000.0,... ): 25.259 KiB       Egamma/stdPhotonAnalyzer
 - DQMHistoSizes: changed ( 1000.0,... ): 5.613 KiB        Egamma/Electrons
 - DQMHistoSizes: changed ( 10024.0,... ): 2.364 KiB        BDHadronTracks/JetContent
 - DQMHistoSizes: changed ( 10024.0,... ): 0.842 KiB        ParticleFlow/PFJetValidation
 - DQMHistoSizes: changed ( 10024.0,... ): 0.451 KiB        ParticleFlow/PFMETValidation
 - DQMHistoSizes: changed ( 20034.0,... ): 25.649 KiB       Egamma/stdPhotonAnalyzerHGCalFromMultiCl
 - DQMHistoSizes: changed ( 20034.0,... ): 7.058 KiB        Egamma/Electrons
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

fabiocos commented:
apparently there was something wrong in my setup, repeating the test

fabiocos commented on Calibration/TkAlCaRecoProducers/python/ALCARECOPromptCalibProdSiStripGainsAAG_cff.py:
```diff
@@ -98,7 +98,6 @@
                                             # 2 provide more detailed output
                                             Frequency = cms.untracked.int32(50),
                                             MEPathToSave = cms.untracked.string('AlCaReco/SiStripGainsAAG'),
-                                            deleteAfterCopy = cms.untracked.bool(True)
```
@tocheng @christopheralanwest @mmusich please suggest how to test this ALCAPROMPT, using ```--dasquery=file dataset=/ZeroBias/Run2016C-SiStripCalMinBiasAfterAbortGap-18Apr2017-v1/ALCARECO run=276097``` in a similar way as done for SiStrpGains does not work:
```
Begin processing the 1st record. Run 276092, Event 32587086, LumiSection 74 on stream 0 at 22-Nov-2019 10:43:57.009 CET
----- Begin Fatal Exception 22-Nov-2019 10:43:57 CET-----------------------
An exception of category 'Configuration' occurred while
   [0] Processing  Event run: 276092 lumi: 74 event: 32587086 stream: 0
   [1] Running path 'pathALCARECOPromptCalibProdSiStripGainsAAG'
   [2] Calling method for module HLTHighLevel/'ALCARECOCalMinBiasFilterForSiStripGainsAAG'
Exception Message:
requested HLT path "pathALCARECOSiStripCalMinBiasAAG" does not exist
----- End Fatal Exception -------------------------------------------------
```

mmusich replied:
Did you change the ALCA producer?

fabiocos replied:
```
cmsDriver.py step3 --datatier ALCARECO --conditions auto:run2_data -s ALCA:PromptCalibProdSiStripGainsAAG --eventcontent ALCARECO -n 1000 --dasquery=file dataset=/ZeroBias/Run2016C-SiStripCalMinBiasAfterAbortGap-18Apr2017-v1/ALCARECO run=276097 ```

mmusich replied:
@fabiocos, the command is correct but the input dataset is not.  
In 2017 we changed 'SiStripCalMinBiasAfterAbortGap' to 'SiStripCalMinBiasAAG' due to limitations in the output dataset name length in dataset storage system.
Because of this the PD you chose is not processable tout-court in recent releases, but needs to have this line https://github.com/cms-sw/cmssw/blob/master/Calibration/TkAlCaRecoProducers/python/ALCARECOPromptCalibProdSiStripGainsAAG_cff.py#L9 adjusted.
To make it work out of the box just switch to a more recent dataset like
 '/StreamExpress/Run2018C-SiStripCalMinBiasAAG-Express-v1/ALCARECO' (is at 'T2_CH_CERN')

fabiocos replied:
@mmusich with ```step3 --processName ALCA2 --datatier ALCARECO --conditions auto:run2_data -s ALCA:PromptCalibProdSiStripGainsAAG --eventcontent ALCARECO -n 1000 --dasquery=file dataset=/StreamExpress/Run2018C-SiStripCalMinBiasAAG-Express-v1/ALCARECO``` I get

![stripaag](https://user-images.githubusercontent.com/4058194/69449283-d9c4ea00-0d5a-11ea-9892-4bdb8fb1290d.png)

which looks to me comparable to your test at the file crossing boundaries

fabiocos commented on Calibration/EcalCalibAlgos/python/ALCARECOPromptCalibProdEcalPedestals_cff.py:
```diff
@@ -27,7 +27,6 @@
                                              # 2 provide more detailed output
                                              Frequency=cms.untracked.int32(50),
                                              MEPathToSave=cms.untracked.string('AlCaReco/EcalPedestalsPCL'),
-                                             deleteAfterCopy=cms.untracked.bool(True)
```
@tocheng @christopheralanwest @schneiml this can be tested using wf 1010.0 . In its step3 the log file is flooded with 

```
Begin processing the 100th record. Run 299149, Event 579095302, LumiSection 408 on stream 3 at 20-Nov-2019 17:35:47.296 CET
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_0' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_1' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_10' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_11' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_12' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_13' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_14' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_15' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_16' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_17' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_18' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_19' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_2' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_20' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_21' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_22' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_23' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_24' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_25' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_26' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_27' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_28' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_29' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_3' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_30' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_31' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_32' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_33' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_34' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_35' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_36' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_37' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_38' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_39' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_4' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_40' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_41' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_42' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_43' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_44' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_45' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_46' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_47' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_48' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_49' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_5' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_50' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_51' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_52' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_53' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_54' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_55' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_56' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_57' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_58' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_59' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_6' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_60' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_61' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_62' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_63' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_64' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_65' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_66' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_67' in 'AlCaReco/EcalPedestalsPCL/EB/0'
DQMStore: WARNING: attempt to remove non-existent monitor element 'eb_68' in 'AlCaReco/EcalPedestalsPCL/EB/0'
(...)
```

Please comment

fabiocos replied:
Repeating the test of @mmusich with wf 1010 step3 running 10000 events:

[ecal.pdf](https://github.com/cms-sw/cmssw/files/3879162/ecal.pdf)

schneiml replied:
Re the message see the comment I made in another location: the code that used to produce them is deleted in this PR.

Re the RSS graph: We see the usual MEtoEDM endJob spike, before as after the PR, doubling the memory use as all histograms are copied into a EDM product. Removing the delete-calls makes it an exact doubling, before it might have been a bit less, as some histos where deleted before saving.

This spike is why "normal" reco jobs switched to DQMIO saving in LS1 (which removes this spike alltogether), but for the ALCA workflows memory use was not considered a problem and they where never migrated.

fabiocos replied:
@schneiml the message is in the default version, this PR removes it, sorry for the possible confusion.

Concerning the RSS plot I do not see anything worrying, the behaviour old vs new looks quite similar.

schneiml replied:
@fabiocos re the message: ok, that is expected, there is nothing deleted any more and ofc there is no reason to warn about deleting non-existing things then.

This *might* explain why the memory use is effectively unaffected: The delete calls were not doing anything to begin with.

schneiml commented on DQMServices/Core/src/DQMStore.cc:
```diff
@@ -2805,60 +2805,6 @@ namespace dqm::dqmstoreimpl {
-    if (pos != data_.end())
-      data_.erase(pos);
-    else if (warning) {
-      std::cout << "DQMStore: WARNING: attempt to remove non-existent"
```
@fabiocos I don't fully understand your comment in a parallel thread, were you testing the old or the new code?

The code that makes the messages you show is deleted here.

fabiocos commented on Calibration/TkAlCaRecoProducers/python/ALCARECOSiStripPCLHistos_cff.py:
```diff
@@ -9,7 +9,6 @@
                                        # 2 provide more detailed output
                                        Frequency = cms.untracked.int32(50),
                                        MEPathToSave = cms.untracked.string('AlCaReco/SiStrip'),
-                                       deleteAfterCopy = cms.untracked.bool(False)
```
running the step2 of wf 1040 with the ```ALCAPRODUCER:@allForExpress``` option:
![express](https://user-images.githubusercontent.com/4058194/69442998-d2e3aa80-0d4d-11ea-9654-9df955b2be69.png)

mmusich replied:
@fabiocos, I am sorry to be picky, but I don't think `@allForExpress` resolves into the `PromptCalibProd...` ALCAPROMPT producer changed here, so I am afraid your plot proves nothing. 
A more interesting test IMHO is to monitor the RSS of step3 of wf 1001.0:
```
cmsDriver.py step3  --datatier ALCARECO --triggerResultsProcess RECO --conditions
auto:run1_data -s ALCAOUTPUT:SiStripCalZeroBias+TkAlMinBias+DtCalib+Hotline+LumiPix
elsMinBias+AlCaPCCZeroBiasFromRECO+AlCaPCCRandomFromRECO,ALCA:PromptCalibProd+Promp
tCalibProdSiStrip+PromptCalibProdSiStripGains+PromptCalibProdSiStripGainsAAG+Prompt
CalibProdSiPixelAli+PromptCalibProdSiPixel --eventcontent ALCARECO -n 100  --filein
  file:step2.root  --fileout file:step3.root  
```

fabiocos replied:
@mmusich as far as I can see in also the Tracer dump it do runs both ```MEtoEDMConvertSiStrip``` and ```pathALCARECOSiStripPCLHistos``` that are also modified in this PR

mmusich replied:
What about `PromptCalipProdSiStripGais...` and `PromptCalibProdSiPixelAli`?

mmusich replied:
Or your test was limited to this particular file change?

fabiocos replied:
@mmusich I believe that we are clearly missing a test that is realistically mimicking what happens nowadays at the T0 , not bits and pieces. My understanding from a discussion with @tocheng is that ```@allForExpress``` should cover all the ALCAPRODUCER available, and I looked at a few ALCAPROMPT touched by this PR separately. 
It would be important to get from ALCA  a realistic workflow on recent data. Anyway I will make a final check following your suggestion (but is this test on very old data representative of what happens in today's situation?)

mmusich replied:
Thanks for the additional tests. I agree about the lack of a suitable relval test testing all the AlCa aspects of Express.
For tracker at least 1001.0 is a reasonable proxy. For the changes proposed here I don't think occupancy or n. of channel differences matter to measure a comparison of the memory consumption (it would be nice to have 1001.0 updated with Run2 input data though).

fabiocos replied:
running wf 1001 on 5000 events, the comparison of RSS used (by a 4-threads jobs) by step3 is

![express2011](https://user-images.githubusercontent.com/4058194/69497904-271f9380-0ee2-11ea-8b61-05d240860be9.png)

fabiocos commented:
+1

schneiml commented:
Thanks everybody for the help with getting this integrated! -- now lets move on to #28379, which I am currently rebasing on top of this.
